legion/engine/physics/colliders/convexcollider.cpp

```diff
@@ -62,14 +63,27 @@ namespace legion::physics
         float aToBEdgeSeperation;
         //log::debug("Edge Check");
         if (PhysicsStatics::FindSeperatingAxisByGaussMapEdgeCheck(this, convexCollider, manifold.transformB, manifold.transformA,
-            edgeRef, edgeInc, edgeNormal, aToBEdgeSeperation) || !edgeRef.ptr)
+            edgeRef, edgeInc, edgeNormal, aToBEdgeSeperation, shouldDebug) || !edgeRef.ptr)
         {
-            //log::debug("aToBEdgeSeperation {} " , aToBEdgeSeperation);
+            //
             manifold.isColliding = false;
+
+            if (shouldDebug)
+            {
+                edgeRef.ptr->DEBUG_drawEdge(manifold.transformB, math::colors::blue, 10.0f);
+                edgeInc.ptr->DEBUG_drawEdge(manifold.transformA, math::colors::red, 10.0f);
+            }
             return;
         }
 
+        if (shouldDebug)
+        {
+            log::debug("-> collision detected for debug ");
+            ARefFace.ptr->DEBUG_DrawFace(manifold.transformA, math::colors::blue, 5.0f);
+            BRefFace.ptr->DEBUG_DrawFace(manifold.transformB, math::colors::red, 5.0f);
+        }
```

Comment:
If you still need these debugs, you can maybe remove the if-statement around them